libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -446,6 +338,124 @@ impl NodeAPI for Greenlight {
         client.key_send(request).await?.into_inner().try_into()
     }
 
+    async fn start(&self) -> Result<()> {
+        self.get_client().await?;
+        Ok(())
+    }
+
+    async fn sweep(
+        &self,
+        to_address: String,
+        fee_rate_sats_per_vbyte: u64,
+    ) -> Result<WithdrawResponse> {
+        let mut client = self.get_client().await?;
+
+        let request = pb::WithdrawRequest {
+            feerate: Some(pb::Feerate {
+                value: Some(pb::feerate::Value::Perkw(fee_rate_sats_per_vbyte * 250)),
+            }),
+            amount: Some(Amount {
+                unit: Some(Unit::All(true)),
+            }),
+            destination: to_address,
+            minconf: None,
+            utxos: vec![],
+        };
+
+        Ok(client.withdraw(request).await?.into_inner())
+    }
+
+    async fn prepare_withdraw(
+        &self,
+        to_address: String,
+        fee_rate_sats_per_vbyte: u32,
+        withdraw_amount_msats: Option<u64>,
```

Comment:
Why do we need the amount? Shouldn't we take it from the on-chain balance?

Reply:
On Breez Mobile, we can choose an amount to withdraw. So I assumed it was part of the prerequisites, wasn't it?

Reply:
> Starting to shape up! It is hard to review though with all the formatting changes.

@ademar111190 FYI: you can solve this by calling `cargo fmt --all` in `sdk-core` before pushing new commits.